SPECIFICATIONS.md

```diff
@@ -683,16 +684,58 @@ is a Biscuit token, that base 64 string should be prefixed with `biscuit:`.
 
 ### Cryptography
 
-Biscuit tokens are based on public key cryptography, with a chain of Ed25519
+Biscuit tokens are based on public key cryptography, with a chain of
 signatures. Each block contains the serialized Datalog, the next public key,
 and the signature by the previous key. The token also contains the private key
 corresponding to the last public key, to sign a new block and attenuate the
 token, or a signature of the last block by the last private key, to seal the
 token.
 
+#### Algorithms
+
+Biscuit supports multiple signature algorithms for it blocks, that can change
+between blocks in one token. The algorithm kind is defined in the `Algorithm`
+enum of the protobuf serialization of the public key. The `nextSecret` field
+in the proof section of the token uses the same algorithm as the `nextKey`
+of the last block.
+
+The following algorithms are supported:
+
+##### Ed25519
+
+The default signature algorithm is Ed25519 as introduced in [Bernstein, Daniel J.;
+Duif, Niels; Lange, Tanja; Schwabe, Peter; Bo-Yin Yang (2012). "High-speed
+high-security signatures" (PDF). Journal of Cryptographic Engineering](https://ed25519.cr.yp.to/ed25519-20110926.pdf)
+and specified in [RFC 8032](https://www.rfc-editor.org/rfc/rfc8032).
+
+The protobuf encoding is defined as follows:
+- `key` field of the `Publickey` message: [compressed Edwards Y format](https://www.rfc-editor.org/rfc/rfc8032#section-5.1.2)
+- `nextSecret` in the `Proof` message: [32 bytes of cryptographically secure random data in little-endian](https://www.rfc-editor.org/rfc/rfc8032#section-5.1.5)
+- `signature` field in `Signature` and `ExternalSignature` messages: [concatenation of R and S values](https://www.rfc-editor.org/rfc/rfc8032#section-5.1.6)
+
+##### ECDSA
+
+Biscuit supports the ECDSA algorithm over the secp256r1 curve as defined in
+[SEC2v1](https://www.secg.org/sec1-v2.pdf), using the SHA-256 hashing
+algorithm. It is recommended to use a deterministic signature algorithm
+version like the one defined in [RFC 6979](https://datatracker.ietf.org/doc/html/rfc6979).
+
+
+The protobuf encoding is defined as follows:
+- `key` field of the `Publickey` message: [SEC1 format, defined in section 2.3.3](https://www.secg.org/sec1-v2.pdf). Allowed prefixes: `02`, `03`, `04`
```

Comment:
Do you really need to support uncompressed public keys? They're just a workaround for long-expired point compression patents.

Reply:
fixed in [bd9901d](https://github.com/biscuit-auth/biscuit/pull/165/commits/bd9901d70a72b3f6c1163114f1eea996ccc289b0)